### ISSUE TITLE
TwigException: Unable to parse 'macro test(param = '')'

### ISSUE THREAD

**Issue by ryuran** [twigjs/twig.js#531]
Hello

Twig.js is unable to parse a macro with default value for a parameter.

```twig
{% macro test(param = '') %} test {% endmacro %}
```

It’s working fine with Twig php.

**Comment by toptalo** [comment]
Macros differ from native PHP functions in a few ways:
- Default argument values are defined by using the `default` filter in the macro body;
- Arguments of a macro are always optional.
- If extra positional arguments are passed to a macro, they end up in the special `varargs` variable as a list of values.

https://twig.symfony.com/doc/2.x/tags/macro.html

So:

```twig
{% macro test(param) %} test {{ param|default('') }} {% endmacro %}
```

**Comment by olets** [comment]
But ryuran is right that despite what's included in the Twig documentation, macros do accept defaults:

```twig
{% macro test(param = 2) %} {{param}} {% endmacro %}

{% import _self as macros %}

{{ macros.test() }}
```

returns `2`

https://twigfiddle.com/dw63fe

**Comment by toptalo** [comment]
Yes it looks like undocument feture

https://github.com/twigphp/Twig/blob/9402071768157c115ebf35195661222bd7ba85cf/lib/Twig/Node/Macro.php#L43

**Comment by ericmorand** [comment]
And it's tested by Twig integration test suite: https://github.com/twigphp/Twig/blob/2.x/test/Twig/Tests/Fixtures/macros/default_values.test

**Comment by dave-irvine** [comment]
Closed by #544

**Comment by BillyWM** [comment]
Does this still work in 2022? Was this implemented correctly in #544? Because I'm still getting a parse error with default arguments but can write a macro with only positional arguments
